### ISSUE TITLE
Chinese .ttf fonts failing with one of two errors

### ISSUE THREAD

**Issue by dhowe** [nodebox/opentype.js#279]
Uncaught Error: Only format 4 cmap tables are supported.
or
Uncaught TypeError: Cannot read property 'glyphIndexMap' of null

**Comment by fdb** [comment]
Both errors have to do with unsupported `cmap` tables.

- The `DFBK1B.TTF` font file has a supported cmap sub-table, but with an unsupported format 0.
- The `heiti.ttf` font file only has a Unicode sub-table (PlatformID 0 EncodingId 3), which we do not support.

I'm not sure about supporting format 0 as used by `DFBK1B.TTF`, since it can only encode 255 characters and seems not to be used very often.

The [Unicode platform](https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6cmap.html) is something we *should* support, as more fonts are using this, including some built-in fonts on OS X.

**Comment by dhowe** [comment]
Indeed, unicode support would be great. We had significant difficulty getting any (what we thought were) standard Chinese fonts on OS X to work with opentype

**Comment by fdb** [comment]
With the introduction of CID-keyed fonts, a lot more of them should now work, although some still miss the correct cmap parsing.

**Comment by weiyunpeng** [comment]
I have also encountered this situation. Is it resolved now?
[1.TTF.zip](https://github.com/opentypejs/opentype.js/files/4244053/1.TTF.zip)

**Comment by antonytse** [comment]
> 
> 
> I have also encountered this situation. Is it resolved now?
> [1.TTF.zip](https://github.com/opentypejs/opentype.js/files/4244053/1.TTF.zip)

This is a font without any Unicode `cmap` tables (platform ID 3 encoding ID 3). Seems the library will not support it at this moment, but I would like to try to make it work for this library.

**Comment by Connum** [comment]
`heiti.ttf` and `DFBK1B.TTF` already work with the current master, `1.TTF` will work as soon as #648 will be implemented
